### PR TITLE
Design Time Host changes to improve socket connection and monitoring

### PR DESCRIPTION
**Pull request by wolfspider**
Original socket connection was exceeding it's timeout checking the connection before the socket was bound. Socket status is checked and logged asynchronously.

### CODE REVIEW

File: src/OmniSharp.Dnx/Utilities/SocketConnection.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                   {
+                       Logger.LogError("Connection Timed Out");
+                       throw new SocketException((int)e.SocketError);
+                   }
+                  break;
+            }
+        }
+        
+}
+
```
Comment (troydai) [inline src/OmniSharp.Dnx/Utilities/SocketConnection.cs]: Please format the code here.

File: src/OmniSharp.Dnx/Utilities/SocketConnection.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+using System.Net.Sockets;
+using System.Threading;
+using Microsoft.Framework.Logging;
+
+namespace OmniSharp
+{
+    public class SocketConnection
+    {
+        public ILogger Logger { get; private set; }
+        public string ContextFlag { get; private set; }
```
Comment (troydai) [inline src/OmniSharp.Dnx/Utilities/SocketConnection.cs]: Both properties are initialized in constructor only. You can remove the `private set;` here.

Comment (troydai) [inline src/OmniSharp.Dnx/Utilities/SocketConnection.cs]: Can `ContextFlag` be a private field only?

File: src/OmniSharp.Dnx/Utilities/SocketConnection.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    
+            Thread.Sleep(500);
+                try
+                {
+                socket.ConnectAsync(connect);                
+                }
+                catch (SocketException)
+                {
+                // this happens when the DTH isn't listening yet
+                Logger.LogVerbose("[SocketConnection]: Socket Connection failed");
```
Comment (troydai) [inline src/OmniSharp.Dnx/Utilities/SocketConnection.cs]: What about LogError?

**Comment by david-driscoll** [comment]
Travis is mean.

**Comment by david-driscoll** [comment]
Travis is being mean, only because the travis.yml is wrong...  I'll fix that tonight, otherwise this looks okay after the feedback.

**Comment by david-driscoll** [comment]
@wolfspider if you could pull in the travis change (it's now in master), it should then pass the build.

**Comment by wolfspider** [comment]
@david-driscoll Thanks, I'll give it a try shortly

**Comment by troydai** [comment]
@wolfspider @david-driscoll since we're moving towards `dotnet-cli` I'm wondering if we still want to take this change.

**Comment by wolfspider** [comment]
@troydai from my perspective I was not able to use VSCode at all without many retries unless I patched this code in. I originally started looking into this because I found one other person who experienced the same problem while looking for answers. The connection time also speeds up when launching VSCode using this code. If the DTH is being replaced altogether in an upcoming release soon then I don't see a need for it however. If its not a common issue for other people then we probably don't need it as I was even expecting more occurrences of the issue to roll in. If there is going to be some time before dotnet-cli lands in VSCode which is the main entry point for the Mac users I think it would be good to have it included.

**Comment by troydai** [comment]
@wolfspider thanks for the clarification. I'm fine with merge this. But DTH is going to be replaced altogether. However I am not able to comment on schedule. You can try out the `dev` branch and you will see it is fairly close.

@david-driscoll how do you like this PR?

**Comment by filipw** [comment]
I think this PR is now obsolete following yesterday's merge of dotnet-cli

**Comment by troydai** [comment]
@wolfspider I'll close this PR since we're moving on to the `dotnet` now. Thanks.

**Pull request closed without merging**